backend/handler/elastic.js

```diff
@@ -0,0 +1,141 @@
+*
+*  You should have received a copy of the GNU Affero General Public License
+*  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+"use strict";
+const { Client } = require("@elastic/elasticsearch");
+const logger = require("../logger");
+const systemLogger = logger.systemLogger;
+const elasticConfig = require("../config").elastic;
+const Elastic = {};
+
+const loginRecordIndex = "io-teamspace-loginrecord";
+const loginRecordMapping = {
+	"username" : { "type": "text" },
+	"loginTime" : { "type": "date" },
+	"ipAddr" : { "type": "ip" },
+	"location.country" : { "type": "text" },
+	"location.city" : { "type": "text" },
+	"referrer" : { "type": "text" },
+	"application.name": { "type": "text" },
+	"application.version": { "type": "text" },
+	"application.type": { "type": "text" },
+	"engine.name": { "type": "text" },
+	"engine.version": { "type": "text" },
+	"os.name": { "type": "text" },
+	"os.version": { "type": "text" },
+	"device": { "type": "text" }
+};
+
+const indicesMappings = [
+	{
+		index: loginRecordIndex,
+		mapping: loginRecordMapping
+	}
+];
+
+const createElasticClient = async () => {
+	if(!elasticConfig) {
+		return;
+	}
+
+	const elasticCredentials = elasticConfig.cloudAuth.split(":");
+	const config = {
+		cloud: {
+			id: elasticConfig.cloudId
+		},
+		auth: {
+			username: elasticCredentials[0],
+			password: elasticCredentials[1]
+		},
+		reload_connections: elasticConfig.reload_connections,
+		maxRetries: elasticConfig.maxRetries,
+		request_timeout: elasticConfig.request_timeout
+	};
+
+	const client = new Client(config);
+	try {
+		await client.cluster.health();
+		systemLogger.logInfo(`Succesfully connected to ${elasticConfig.cloudId.trim()}`);
+		await establishIndices(client);
+	} catch (err) {
+		systemLogger.logError("Health check failed on elastic connection, please check settings.");
+		return;
+	}
+
+	return client;
+};
+
+const establishIndices = async (elasticClient)=>{
+	indicesMappings.forEach(async (indexMapping) => {
+		const { body } = await elasticClient.indices.exists({ index: indexMapping.index });
+		if (!body) {
+			await elasticClient.indices.create({index: indexMapping.index });
+			systemLogger.logInfo(`Created index ${indexMapping.index}`);
+			if (indexMapping.mapping) {
+				await elasticClient.indices.putMapping({
+					index: indexMapping.index,
+					body: { properties: indexMapping.mapping }
+				});
+			}
+			systemLogger.logInfo(`Created mapping ${indexMapping.index}`);
+		}
+	});
+};
+
+const elasticClientPromise = createElasticClient();
+
+const createElasticRecord = async (elasticIndex, elasticBody, id) => {
+	try {
+		const elasticClient = await elasticClientPromise;
+
+		if (elasticBody) {
+			await elasticClient.index({
+				index: elasticIndex,
+				id: id,
+				refresh: true,
+				body: elasticBody
+			});
+			systemLogger.logInfo(`created doc ${elasticIndex} ${Object.values(elasticBody).toString()}`);
```

Comment:
probably makes more sense to just do JSON.stringify here? 

```suggestion
			systemLogger.logInfo(`created doc ${elasticIndex} ${JSON.stringify(elasticBody)}`);
```